src/components/List/List.vue

```diff
@@ -1,14 +1,104 @@
 <template>
-	<ul>
+	<ul ref="contentRef" @focusin="onFocusin" @focusout="onFocusout" @focus="onFocus">
 		<slot></slot>
 	</ul>
 </template>
 <script lang="ts">
-import Vue from 'vue';
+import Vue, { onMounted, onUnmounted, ref } from 'vue';
 
 export default Vue.extend({
 	name: 'farm-list',
-	setup() {},
+	setup(_, { emit }) {
+		const contentRef = ref<HTMLElement>();
+		const isFocused = ref(false);
+
+		onMounted(() => {
+			contentRef.value.querySelectorAll('[tabindex]:not([tabindex="-1"]), li').forEach(tag =>
+				tag.addEventListener('keydown', (e: KeyboardEvent) => {
+					e.preventDefault();
+					if (isFocused.value) {
+						emit('keydown', e);
+					}
+				})
+			);
+		});
+
+		onUnmounted(() => {
+			contentRef.value.querySelectorAll('[tabindex]:not([tabindex="-1"]), li').forEach(tag =>
+				tag.removeEventListener('keydown', (e: KeyboardEvent) => {
+					e.preventDefault();
+					if (isFocused.value) {
+						emit('keydown', e);
+					}
+				})
+			);
+		});
+
+		function focus(location?: 'next' | 'prev' | 'first' | 'last') {
+			if (!contentRef.value) return;
+
+			const focusable = [
+				...contentRef.value.querySelectorAll('[tabindex]:not([tabindex="-1"]), li'),
+			].filter(el => !el.hasAttribute('disabled')) as HTMLElement[];
+
+			const idx = focusable.indexOf(document.activeElement as HTMLElement);
+
+			if (!location) {
+				if (!contentRef.value.contains(document.activeElement)) {
+					focusable[0].focus();
+				}
+			} else if (location === 'first') {
+				let savedTabIndex = focusable[0].getAttribute('tabindex');
+
+				focusable[0].setAttribute('tabindex', '-1');
+				focusable[0].focus();
+				focusable[0].setAttribute('tabindex', savedTabIndex);
+			} else if (location === 'last') {
+				focusable.at(-1).focus();
+			} else {
+				let el;
+				let idxx = idx;
+				const inc = location === 'next' ? 1 : -1;
+
+				do {
+					idxx += inc;
+					el = focusable[idxx];
+					console.log('el', el);
+				} while ((!el || el.offsetParent == null) && idxx < focusable.length && idxx >= 0);
+				if (el) {
+					el.focus();
+				} else {
+					focus(location === 'next' ? 'first' : 'last');
+				}
+			}
+		}
+
+		function onFocusin() {
+			isFocused.value = true;
+		}
+
+		function onFocusout() {
+			isFocused.value = false;
+		}
+
+		function onFocus(e: FocusEvent) {
+			if (
```

Comment:
Esse if está fazendo uma lógica assim:

Se não CondiçãoA e não CondiçãoB, faça X.

È mais fácil cognitivamente falando usar o valor verdadeiro na comparação e não a negação:
Se não for (CondiçãoA ou CondiçãoB), faça X.

Reply:
Feito as alterações.